src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -30,6 +30,7 @@ public function __construct(Environment $twig)
     public function render(object $component, string $template): string
     {
         // TODO: Self-Rendering components?
-        return $this->twig->render($template, ['this' => $component]);
+        $templateName = $component->getTemplate() ?? $template;     
```

Comment:
I don't understand how this works (I could be totally missing something). The `$component` here is your component instance. But your component does not use or extend `AsTwigComponent`: that's an attribute on your component.

Reply:
I didn't need to put `function getTemplate()` in `AsTwigComponent`. It works without.